packages/tangle/timemanager.go

```diff
@@ -117,33 +140,59 @@ func (t *TimeManager) synced() bool {
 	return clock.Since(t.lastConfirmedMessage.Time) < t.tangle.Options.SyncTimeWindow
 }
 
+// checks whether the synced state needs to be updated and if so,
+// triggers a corresponding event reflecting the new state.
+func (t *TimeManager) updateSyncedState() {
+	t.lastSyncedMutex.Lock()
+	defer t.lastSyncedMutex.Unlock()
+	if newSynced := t.synced(); t.lastSynced != newSynced {
+		t.lastSynced = newSynced
+		// trigger the event inside the lock to assure that the status is still correct
+		t.Events.SyncChanged.Trigger(&SyncChangedEvent{Synced: newSynced})
+	}
+}
+
 // updateTime updates the last confirmed message.
 func (t *TimeManager) updateTime(marker markers.Marker, newLevel int, transition events.ThresholdEventTransition) {
```

Comment:
Is it an issue, if `updateTime` gets called after `Shutdown` do we need to check for that or can it be ignored?

Reply:
Since the `TimeManager` gets shutdown after the components triggering the `MarkerConfirmation` event of the Tangle, `updateTime` should not be called. But even if it would, it would only cause the persisted "last seen confirmed msg" to be inconsistent with what was actually lastly confirmed.